### PR TITLE
Integration specs for views and fix n+1 problems.

### PR DESCRIPTION
**Pull request by kessie2862**
This `PR` addresses the following changes according to the project requirements.

## `Changes Made:`
- [ ] Solved the `N+1` problem when fetching all posts and their comments for a user. See the screenshot below.

### Before
Before

### After
After

### Integration Tests:

Integration tests using `Capybara` were added to validate the application's functionality and UI flow according to the project requirements.

### CODE REVIEW

File: spec/system/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    it "Should see a button that lets me view all of a user's posts." do
+      expect(page).to have_selector("a[href='#{user_posts_path(@user)}']")
+    end
+  end
+
+  describe 'User interaction testing' do
+    before(:each) do
+      visit user_path(@user)
+    end
+
```
Comment (Davidosky007) [inline spec/system/user_show_spec.rb:51]: * Kindly add tests for this scenario as [required](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20a%20user%27s%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page.) `When I click a user's post, it redirects me to that post's show page.
`

File: spec/system/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    it 'Display button to like the post' do
+      expect(page).to have_button('Like Post')
+    end
+
+    it 'like the post' do
+      click_button 'Like Post'
+
+      expect(page).to have_current_path(%r{/users/\d+/posts/\d+})
+
+      expect(page).to have_content('You just liked this post')
```
Comment (Davidosky007) [inline spec/system/post_show_spec.rb:44]: * Kindly add tests for these scenarios as they are [required ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20post,the%20comment%20each%20commentor%20left.)
```
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.
```

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @kessie2862 @ichsansandy 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

* All tests passed 👍 
* Well-written PR 👍 
* No linter errors 👍 
* N+1 problems solved 👍 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/system/post_show_spec.rb (hunk 2 above)
Comment (kessie2862) [reply spec/system/post_show_spec.rb:44]: Added all those 3 additional tests. Thanks

File: spec/system/user_show_spec.rb (hunk 1 above)
Comment (kessie2862) [reply spec/system/user_show_spec.rb:51]: Added this test as well. Thanks

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @kessie2862 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Rails testing features ✔️  
- [x] Kudus on meeting the requirements as  required  ✔️

### Confirmed view:
image


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**